src/engine/controls/loopingcontrol.cpp

```diff
@@ -1277,24 +1277,25 @@ double LoopingControl::seekInsideAdjustedLoop(
     double new_loop_size = new_loop_out - new_loop_in;
     DEBUG_ASSERT(new_loop_size > 0);
     double adjusted_position = currentSample;
-    while (adjusted_position > new_loop_out) {
-        adjusted_position -= new_loop_size;
+    if (adjusted_position > new_loop_out) {
+        double adjust_steps = ceil((adjusted_position - new_loop_out) / new_loop_size);
+        adjusted_position -= adjust_steps * new_loop_size;
```

Comment:
```
const double after_loop_out = fmod(adjusted_position - new_loop_out, new_loop_size);
if (after_loop_out > 0) {
    adjusted_position = new_loop_out - (new_loop_size - after_loop_out);
} else {
    adjusted_position = new_loop_out;
}
```

Reply:
It can happen that we overshoot more than one loop size. That's why we need the ceil() call and the multiplier. 
I will add a comment.

Reply:
That's the reason why my proposal requires a conditional in addition to `fmod`.

Reply:
Ah Ok. The variable name "after_loop_out" was misleading. Is the new comment sufficient?

Reply:
I don't insist on my solution.